tests/assembly/targets/asmjs_unknown_emscripten.rs

```diff
@@ -0,0 +1,17 @@
+// assembly-output: emit-asm
+// compile-flags: --target asmjs-unknown-emscripten
+// needs-llvm-components: webassembly
+
+#![feature(no_core, lang_items)]
+#![no_std]
+#![no_core]
+#![crate_type = "lib"]
+
+#[lang = "sized"]
+trait Sized {}
+
+pub fn test() -> u8 {
+    42
+}
+
+// CHECK: .section
```

Comment:
Being able to write an assembly test here does not mean that this target generates the **correct** code: the object code this target currently generates is instead _wasm_, which allows it to pass the "does it generate assembly" question, but will not actually be asm.js (and makes the target redundant with wasm32-unknown-emscripten).